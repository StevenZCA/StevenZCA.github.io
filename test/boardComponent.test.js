import React from 'react';
import { shallow } from 'enzyme';
import BoardComponent from '../src/js/ES6Modules/boardComponentModule';

describe('Test for GameComponent', ()=> {

    const isValidInput = (values)=>{
        const arr = Array.from(values);
        const isValidChar = arr.every(ele=>
            ['1','2','3','4','5','6','7','8','9','.'].indexOf(ele)>=0
        );
        const hasNumber = arr.some(ele=>
            ['1','2','3','4','5','6','7','8','9'].indexOf(ele)>=0);
        return hasNumber && isValidChar;
    }


    const isEmptyInput = (values)=>{
        const arr = Array.from(values);
        return arr.every(ele=>ele === '.');
    }


    it('test rendering of the BoardComponent that is not Interactive', ()=> {
        const component = shallow(<BoardComponent/>)
        expect(component.find('.board__squareRows__square').length).toEqual(81);
        expect(component).toMatchSnapshot() 
        expect(isEmptyInput(component.instance().state.values)).toEqual(true);
    })

    it('test rendering of the BoardComponent that is interactive', ()=> {
        const component = shallow(<BoardComponent isInteractive={true}/>)
        expect(component.find('.board__squareRows__square').length).toEqual(81);
        //expect(component).toMatchSnapshot(); we don't have a fixed spec since have a random initial set up
        //initial status, the board should have a set up
        expect(isValidInput(component.instance().state.values)).toEqual(true);
    })

    it('test click the auto pick button', ()=> {
        const component = shallow(<BoardComponent isInteractive={true}/>)
        const button = component.find('.autoPick input.board__controls__inputButton');
        button.simulate('click');
        expect(isValidInput(component.instance().state.values)).toEqual(true);
    })

    it('test click the reset button', ()=> {
        const component = shallow(<BoardComponent isInteractive={true}/>)
        const button = component.find('#sudokuResetBtn');
        expect(isValidInput(component.instance().state.values)).toEqual(true);
        button.simulate('click');
        expect(isEmptyInput(component.instance().state.values)).toEqual(true);
    })


});