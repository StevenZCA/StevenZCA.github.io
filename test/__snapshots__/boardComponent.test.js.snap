// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for GameComponent test rendering of the BoardComponent that is not Interactive 1`] = `
<div
  className="board"
>
  <div>
    <div
      className="board__squareRows"
      key="0"
    >
      <div
        key="A1"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="A1"
          index={0}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="A2"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="A2"
          index={1}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="A3"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="A3"
          index={2}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="A4"
      >
        <input
          className="board__squareRows__square"
          id="A4"
          index={3}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="A5"
      >
        <input
          className="board__squareRows__square"
          id="A5"
          index={4}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="A6"
      >
        <input
          className="board__squareRows__square"
          id="A6"
          index={5}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="A7"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="A7"
          index={6}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="A8"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="A8"
          index={7}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="A9"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="A9"
          index={8}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="board__squareRows"
      key="1"
    >
      <div
        key="B1"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="B1"
          index={9}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="B2"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="B2"
          index={10}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="B3"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="B3"
          index={11}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="B4"
      >
        <input
          className="board__squareRows__square"
          id="B4"
          index={12}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="B5"
      >
        <input
          className="board__squareRows__square"
          id="B5"
          index={13}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="B6"
      >
        <input
          className="board__squareRows__square"
          id="B6"
          index={14}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="B7"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="B7"
          index={15}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="B8"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="B8"
          index={16}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="B9"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="B9"
          index={17}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="board__squareRows"
      key="2"
    >
      <div
        key="C1"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="C1"
          index={18}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="C2"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="C2"
          index={19}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="C3"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="C3"
          index={20}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="C4"
      >
        <input
          className="board__squareRows__square"
          id="C4"
          index={21}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="C5"
      >
        <input
          className="board__squareRows__square"
          id="C5"
          index={22}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="C6"
      >
        <input
          className="board__squareRows__square"
          id="C6"
          index={23}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="C7"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="C7"
          index={24}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="C8"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="C8"
          index={25}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="C9"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="C9"
          index={26}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="board__squareRows"
      key="3"
    >
      <div
        key="D1"
      >
        <input
          className="board__squareRows__square"
          id="D1"
          index={27}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="D2"
      >
        <input
          className="board__squareRows__square"
          id="D2"
          index={28}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="D3"
      >
        <input
          className="board__squareRows__square"
          id="D3"
          index={29}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="D4"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="D4"
          index={30}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="D5"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="D5"
          index={31}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="D6"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="D6"
          index={32}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="D7"
      >
        <input
          className="board__squareRows__square"
          id="D7"
          index={33}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="D8"
      >
        <input
          className="board__squareRows__square"
          id="D8"
          index={34}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="D9"
      >
        <input
          className="board__squareRows__square"
          id="D9"
          index={35}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="board__squareRows"
      key="4"
    >
      <div
        key="E1"
      >
        <input
          className="board__squareRows__square"
          id="E1"
          index={36}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="E2"
      >
        <input
          className="board__squareRows__square"
          id="E2"
          index={37}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="E3"
      >
        <input
          className="board__squareRows__square"
          id="E3"
          index={38}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="E4"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="E4"
          index={39}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="E5"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="E5"
          index={40}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="E6"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="E6"
          index={41}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="E7"
      >
        <input
          className="board__squareRows__square"
          id="E7"
          index={42}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="E8"
      >
        <input
          className="board__squareRows__square"
          id="E8"
          index={43}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="E9"
      >
        <input
          className="board__squareRows__square"
          id="E9"
          index={44}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="board__squareRows"
      key="5"
    >
      <div
        key="F1"
      >
        <input
          className="board__squareRows__square"
          id="F1"
          index={45}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="F2"
      >
        <input
          className="board__squareRows__square"
          id="F2"
          index={46}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="F3"
      >
        <input
          className="board__squareRows__square"
          id="F3"
          index={47}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="F4"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="F4"
          index={48}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="F5"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="F5"
          index={49}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="F6"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="F6"
          index={50}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="F7"
      >
        <input
          className="board__squareRows__square"
          id="F7"
          index={51}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="F8"
      >
        <input
          className="board__squareRows__square"
          id="F8"
          index={52}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="F9"
      >
        <input
          className="board__squareRows__square"
          id="F9"
          index={53}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="board__squareRows"
      key="6"
    >
      <div
        key="G1"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="G1"
          index={54}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="G2"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="G2"
          index={55}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="G3"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="G3"
          index={56}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="G4"
      >
        <input
          className="board__squareRows__square"
          id="G4"
          index={57}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="G5"
      >
        <input
          className="board__squareRows__square"
          id="G5"
          index={58}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="G6"
      >
        <input
          className="board__squareRows__square"
          id="G6"
          index={59}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="G7"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="G7"
          index={60}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="G8"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="G8"
          index={61}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="G9"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="G9"
          index={62}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="board__squareRows"
      key="7"
    >
      <div
        key="H1"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="H1"
          index={63}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="H2"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="H2"
          index={64}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="H3"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="H3"
          index={65}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="H4"
      >
        <input
          className="board__squareRows__square"
          id="H4"
          index={66}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="H5"
      >
        <input
          className="board__squareRows__square"
          id="H5"
          index={67}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="H6"
      >
        <input
          className="board__squareRows__square"
          id="H6"
          index={68}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="H7"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="H7"
          index={69}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="H8"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="H8"
          index={70}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="H9"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="H9"
          index={71}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="board__squareRows"
      key="8"
    >
      <div
        key="I1"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="I1"
          index={72}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="I2"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="I2"
          index={73}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="I3"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="I3"
          index={74}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="I4"
      >
        <input
          className="board__squareRows__square"
          id="I4"
          index={75}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="I5"
      >
        <input
          className="board__squareRows__square"
          id="I5"
          index={76}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="I6"
      >
        <input
          className="board__squareRows__square"
          id="I6"
          index={77}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="I7"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="I7"
          index={78}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="I8"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="I8"
          index={79}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        key="I9"
      >
        <input
          className="board__squareRows__square board__squareRows__square--highlighted"
          id="I9"
          index={80}
          maxLength="1"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;
